packages/astro/src/core/config/schema.ts

```diff
@@ -303,7 +305,14 @@ export const AstroConfigSchema = z.object({
 	markdown: z
 		.object({
 			syntaxHighlight: z
-				.union([z.literal('shiki'), z.literal('prism'), z.literal(false)])
+				.union([
+					z.object({
+						type: highlighterTypesSchema,
+						excludeLangs: z.array(z.string()).optional(),
```

Comment:
We need to add a default value. In line 316, you have an example of how we do it. 

```js
.default(ASTRO_CONFIG_DEFAULTS.markdown.syntaxHighlight)
```